.changeset/early-files-shave.md

```diff
@@ -0,0 +1,22 @@
+---
+'@commercetools-backend/express': patch
+'@commercetools-website/custom-applications': patch
+---
+
+Update express auth verifier to allow it to run in different serverless environments.
+
+In order to determine the 'audience' for JWT validation, the auth verifier needs to resolve the request url.
+By default, the verifier assumes it receives a Node.js IncomingMessage object as its first parameter.
+This default behaviour might not work for every scenario as, for example, AWS lambda functions, where that first parameter is an Event object.
+(Reference: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html)
+To handle these situations, this option allows the delegation of the URL resolution from the request-like object provided to the middleware as its first parameter.
+
+Example for AWS Lambda Funtions:
+
+```
+const sessionAuthVerifier = createSessionAuthVerifier({
+  audience: 'https://my-api-server.com',
+  issuer: CLOUD_IDENTIFIERS.GCP_EU,
+  getRequestUrl: (event) => `${event.rawPath}${event.rawQueryString ? '?' + event.rawQueryString : ''}`
+});
+```
```

Comment:
Since we have already documented this in the docs, we can keep the changelog short and link to the docs.

```suggestion
Allow to map a `request` URL (URI path + query string) using the `getRequestUrl` function in case the `request` object does not contain either a `originalUrl` or `url` properties.

More info at https://docs.commercetools.com/custom-applications/concepts/integrate-with-your-own-api#validating-the-json-web-token.
```

Reply:
Updated in this commit e4be0ae17a4678626bf426a1c7ab237e458a00d1